crates/virtio-queue/src/lib.rs

```diff
@@ -179,6 +182,12 @@ pub trait QueueStateT: for<'a> QueueStateGuard<'a> {
     /// Return the index of the next entry in the available ring.
     fn next_avail(&self) -> u16;
 
+    /// Return the index for the next descriptor in the used ring.
+    fn next_used(&self) -> u16;
+
     /// Set the index of the next entry in the available ring.
     fn set_next_avail(&mut self, next_avail: u16);
+
+    /// Set the index for the next descriptor in the used ring.
+    fn set_next_used(&mut self, next_used: u16);
```

Comment:
Just curious, I had a look at how you are using these in CH, and I was wondering if this one for example would still be needed if you'd switch to the upstream `QueueState`. I'm not very familiar with PCI.

Reply:
We should also add support for the `Versionize` queue state in the near future, based on the agreed proposal from [here](https://github.com/rust-vmm/community/pull/118).

Reply:
Yes I think once QueueState will be derived with Versionize, Cloud Hypervisor will be able to rely fully on it rather than redefining locally. But for now there's no other way.